lark/common.py

```diff
@@ -1,14 +1,29 @@
 from .utils import Serialize
-from .lexer import TerminalDef
+from .lexer import TerminalDef, Token
 
 ###{standalone
+from types import ModuleType
+from typing import Any, Callable, Collection, Dict, Optional, TYPE_CHECKING
 
+if TYPE_CHECKING:
+    from .lark import PostLex
```

Comment:
I think we can just move the definition of `PostLex` into  `common.py`. It fits better here anyway.

Reply:
Why not `lexer.py`?

Btw this is an opportunity to change the postlex interface completely, if we wish to.